docs/intro/index.md

```diff
@@ -4,25 +4,92 @@
 
 This is a work in progress.
 
+We'd love to hear your feedback! If you find any errors or have suggestions for
+the book, please [file an
+issue](https://github.com/ahrefs/melange-for-react-devs/issues) or ping @feihong
+on the [#melange channel in the Reason Discord](https://discord.gg/reasonml).
+
 :::
 
-## Motivation
+## What is Melange?
+
+Melange is a collection of tools that bring the robustness of a mature,
+statically-typed multi-paradigm programming language to the JavaScript
+ecosystem. That language is OCaml, a language invented in the 1990s which has
+been battle-tested by industry stalwarts. The heart of Melange is a compiler
+that translates OCaml to human-readable JavaScript and built-in language
+constructs for zero-cost interoperation with JavaScript.
+
+## Why OCaml?
```

Comment:
Same discussion as ever, but "Why OCaml?" when you will write the book in reason is strange, confusing

Reply:
It seems to me that most of the benefits of Melange come from OCaml, and the book does teach OCaml (using Reason syntax).

Reply:
Oh wait, do you mean Melange in the sense of the subset of OCaml language supported by Melange?

Reply:
To add my 2 cents: It might be beneficial to just add a quick note that communicates that:

Ocaml = general purpose programming language developed in the 90s (and from my understanding, derived from ML
ReasonML = a set of Ocaml extensions/different way of writing OCaml (this is my understanding as someone with lots of functional JavaScript experience, but not a whole lot of ML experience)

Also note that while we are talking about Ocaml here, the reader can expect to see a lot more ReasonML later.